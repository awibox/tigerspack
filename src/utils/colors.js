export const colors = {
  primary: {
    color: '#3F88C5',
    text: '#fff',
    hoverColor: '#619DCF',
    hoverText: '#fff',
  },
  yellow: {
    color: '#FDE74C',
    text: '#304355',
    hoverColor: '#FDEF8D',
    hoverText: '#596977',
  },
  danger: {
    color: '#D15130',
    text: '#fff',
    hoverColor: '#E76846',
    hoverText: '#fff',
  },
  warning: {
    color: '#FA7921',
    text: '#fff',
    hoverColor: '#FB9D5D',
    hoverText: '#fff',
  },
  success: {
    color: '#9BC53D',
    text: '#fff',
    hoverColor: '#B6D471',
    hoverText: '#fff',
  },
  info: {
    color: '#5BC0EB',
    text: '#fff',
    hoverColor: '#87D1F0',
    hoverText: '#fff',
  },
  dark: {
    color: '#304355',
    text: '#fff',
    hoverColor: '#596977',
    hoverText: '#fff',
  },
  light: {
    color: '#E5E8EC',
    text: '#304355',
    hoverColor: '#EEF0F2',
    hoverText: '#596977',
  },
};

export const colorsArray = ['primary', 'danger', 'success'];
