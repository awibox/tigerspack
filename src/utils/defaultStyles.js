export const defaultStyles = {
  animation: 'all .4s ease',
  indent: 15,
  buttonSize: 3,
  borderRadius: 3,
  mediumFontSize: '14px',
};

export const setDefaultStyles = (newStyles) => {
  Object.keys(newStyles).forEach((style) => {
    defaultStyles[style] = newStyles[style];
  });
};
