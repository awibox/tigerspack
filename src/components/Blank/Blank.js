import React from 'react';
import PropTypes from 'prop-types';
import { colors } from '../../utils/colors';

const Blank = (props) => {
  const {
    border,
    children,
    shadow,
    indent,
    rounded,
    padding,
    ...otherProps
  } = props;

  const blankStyles = {
    background: '#fff',
    border: border && `1px solid ${colors[border].color}`,
    boxShadow: !border && `0 0 ${shadow}px rgba(0, 0, 0, 0.12), 0 ${shadow / 2}px ${shadow}px rgba(0, 0, 0, 0.24)`,
    boxSizing: 'border-box',
    transition: 'all 0.3s cubic-bezier(.25, .8, .25, 1)',
    borderRadius: `${rounded}px`,
    padding: `${padding}px`,
    marginBottom: `${indent}px`,
    display: 'flex',
    width: '100%',
  };
  return (
    <div css={blankStyles} {...otherProps}>
      {children}
    </div>
  );
};

Blank.propTypes = {
  border: PropTypes.string,
  children: PropTypes.any.isRequired,
  indent: PropTypes.number,
  shadow: PropTypes.number,
  rounded: PropTypes.number,
  padding: PropTypes.number,
};

Blank.defaultProps = {
  padding: 15,
  rounded: 3,
  shadow: 3,
  indent: 0,
};

export default Blank;
