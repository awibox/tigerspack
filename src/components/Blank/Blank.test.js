import React from 'react';
import { shallow } from 'enzyme';
import Blank from './Blank';
import { colors } from '../../utils/colors';

const testValue = 1;
const testColor = 'primary';

describe('Blank', () => {
  it('Default Blank', () => {
    const component = shallow(<Blank>Test Blank</Blank>);
    expect(component).toMatchSnapshot();
  });
  it('Grid shadow', () => {
    const component = shallow(<Blank shadow={testValue}>Test Blank</Blank>);
    expect(component.get(0).props.css.boxShadow).toEqual(`0 0 ${testValue}px rgba(0, 0, 0, 0.12), 0 ${testValue / 2}px ${testValue}px rgba(0, 0, 0, 0.24)`);
  });
  it('Grid rounded', () => {
    const component = shallow(<Blank rounded={testValue}>Test Blank</Blank>);
    expect(component.get(0).props.css.borderRadius).toEqual(`${testValue}px`);
  });
  it('Grid indent', () => {
    const component = shallow(<Blank indent={testValue}>Test Blank</Blank>);
    expect(component.get(0).props.css.marginBottom).toEqual(`${testValue}px`);
  });
  it('Grid padding', () => {
    const component = shallow(<Blank padding={testValue}>Test Blank</Blank>);
    expect(component.get(0).props.css.padding).toEqual(`${testValue}px`);
  });
  it('Grid border', () => {
    const component = shallow(<Blank border={testColor}>Test Blank</Blank>);
    expect(component.get(0).props.css.border).toEqual(`1px solid ${colors[testColor].color}`);
  });
});
