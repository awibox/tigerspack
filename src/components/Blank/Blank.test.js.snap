// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blank Default Blank 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  css={
    Object {
      "background": "#fff",
      "border": undefined,
      "borderRadius": "3px",
      "boxShadow": "0 0 3px rgba(0, 0, 0, 0.12), 0 1.5px 3px rgba(0, 0, 0, 0.24)",
      "boxSizing": "border-box",
      "display": "flex",
      "marginBottom": "0px",
      "padding": "15px",
      "transition": "all 0.3s cubic-bezier(.25, .8, .25, 1)",
      "width": "100%",
    }
  }
>
  Test Blank
</EmotionCssPropInternal>
`;
